src/test/ui/async-await/issue-70935-complex-spans.drop_tracking.stderr

```diff
@@ -22,7 +22,7 @@ LL |   async fn baz<T>(_c: impl FnMut() -> T) where T: Future<Output=()> {
 LL | |
 LL | | }
    | |_^
-   = note: required because it captures the following types: `ResumeTy`, `impl Future<Output = ()>`, `()`
+   = note: required because it captures the following types: `ResumeTy`, `impl for<'r, 's, 't0> Future<Output = ()>`, `()`
```

Comment:
hmm, is this change expected? I guess it doesn't really matter since I'm going to remove it in https://github.com/rust-lang/rust/pull/98754, but I wonder if it can come up in diagnostics for generators or something like that?

Reply:
well so the only reason we weren't printing this in the first place was because of another bug... the bound lifetimes are there though

i mean i guess we could filter them out or something 🤷

Reply:
hmm, sure - I think what's confusing is that the lifetimes don't appear in the `impl Future` pretty printing, only the `for` binder.

maybe that's ok though and this gives a hint of which lifetimes are captured? I don't feel super strongly either way